app_grim_reefer/src/data_storage.c

```diff
@@ -0,0 +1,110 @@
+    }                                                                                                                  \
+                                                                                                                       \
+    event.state = K_POLL_STATE_NOT_READY;
+
+#define OPEN_OR_FAIL(name, filename)                                                                                   \
+    struct fs_file_t name;                                                                                             \
+    {                                                                                                                  \
+        fs_file_t_init(&name);                                                                                         \
+        int ret = fs_open(&name, filename, FS_O_RDWR | FS_O_CREATE);                                                   \
+                                                                                                                       \
+        if (ret < 0) {                                                                                                 \
+            LOG_ERR("Error opening %s. %d", filename, ret);                                                            \
+            k_event_post(&storage_setup_finished, STORAGE_SETUP_FAILED_EVENT);                                         \
+            return;                                                                                                    \
+        }                                                                                                              \
+    }
+
+void storage_thread_entry_point(void *, void *, void *) {
+    struct fast_data fast_dat;
+    struct slow_data slow_dat;
+    struct adc_data adc_dat;
+    enum flight_event event;
+    int ret;
+
+    OPEN_OR_FAIL(fast_file, FAST_FILENAME);
+    OPEN_OR_FAIL(slow_file, SLOW_FILENAME);
+    OPEN_OR_FAIL(adc_file, ADC_FILENAME);
+
+    k_event_post(&storage_setup_finished, STORAGE_SETUP_SUCCESS_EVENT);
+
+    for (;;) {
+        k_poll(events, NUM_EVENTS, K_FOREVER);
+
+        // Slow data
+        CHECK_AND_STORE(events[0], slow_dat, slow_file, SLOW_FILENAME)
+        // ADC data
+        CHECK_AND_STORE(events[1], adc_dat, adc_file, ADC_FILENAME)
+        // Fast data
+        CHECK_AND_STORE(events[2], fast_dat, fast_file, FAST_FILENAME)
+
+        if (events[3].state == K_POLL_STATE_MSGQ_DATA_AVAILABLE) {
+            k_msgq_get(events[3].msgq, &event, K_NO_WAIT);
+            if (event == flight_event_main_shutoff) {
+                break;
+            }
+        }
+    }
+    LOG_INF("Flight over. Saving files...");
+    ret = fs_close(&fast_file);
+    if (ret < 0) {
+        LOG_ERR("Failed to fast file. Uh oh");
+    }
+    ret = fs_close(&slow_file);
+    if (ret < 0) {
+        LOG_ERR("Failed to slow file. Uh oh");
+    }
+    ret = fs_close(&adc_file);
+    if (ret < 0) {
+        LOG_ERR("Failed to adc file. Uh oh");
```

Comment:
Would be nice to log the return value

Reply:
Also NIT, but it looks like you forgot to have close so "Failed to **close** XXXX file"